src/box/lua/space.cc

```diff
@@ -417,6 +417,36 @@ luaT_add_index_parts_methods(struct lua_State *L, const struct key_def *key_def)
 	lua_setmetatable(L, -2);
 }
 
+/**
+ * Push array with covered fields on Lua stack. Implicitly covered fields are
+ * added, the array is sorted.
+ */
+static void
+luaT_push_covered_fields(struct lua_State *L, struct index_def *def)
+{
+	struct region *gc = &fiber()->gc;
+	size_t gc_svp = region_used(gc);
+	uint32_t field_count = def->opts.covered_field_count +
+			       def->cmp_def->part_count;
+	uint32_t *fields = xregion_alloc_array(gc, typeof(*fields),
+					       field_count);
+	uint32_t j = def->opts.covered_field_count;
+	memcpy(fields, def->opts.covered_fields, sizeof(*fields) * j);
+	for (uint32_t i = 0; i < def->cmp_def->part_count; i++) {
+		if (!index_opts_is_covered(&def->opts,
```

Comment:
AFAIU indexed fields are excluded from the covered fields array during normalization, which means that this check isn't required.